lib/ui/window/pointer_data_packet_converter.cc

```diff
@@ -0,0 +1,288 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/window/pointer_data_packet_converter.h"
+#include "flutter/fml/logging.h"
+
+#include <string.h>
+
+namespace flutter {
+
+PointerDataPacketConverter::PointerDataPacketConverter() : pointer_(0) {}
+
+PointerDataPacketConverter::~PointerDataPacketConverter() = default;
+
+std::unique_ptr<PointerDataPacket> PointerDataPacketConverter::Expand(
+    std::unique_ptr<PointerDataPacket> packet) {
+  size_t kBytesPerPointerData = kPointerDataFieldCount * kBytesPerField;
+  auto buffer = packet->data();
+  size_t buffer_length = buffer.size();
+
+  std::vector<PointerData> expended_pointers;
+  // Expands each pointer data in the buffer and stores it in the
+  // expended_pointers.
+  for (size_t i = 0; i < buffer_length / kBytesPerPointerData; i++) {
+    PointerData pointer_data;
+    memcpy(&pointer_data, &buffer[i * kBytesPerPointerData],
```

Comment:
Can we just default a copy constructor in `PointerData` instead of resorting to the memcpy here please?